YACReaderLibrary/db/comic_model.cpp

```diff
@@ -595,59 +596,25 @@ void ComicModel::setupReadingModelData(const QString &databasePath)
     endResetModel();
 }
 
-void ComicModel::setupModelData(const SearchModifiers modifier, const QString &filter, const QString &databasePath)
+void ComicModel::setModelData(QList<ComicItem *> *data, const QString &databasePath)
 {
-    beginResetModel();
-    qDeleteAll(_data);
-    _data.clear();
     _databasePath = databasePath;
-    QString connectionName = "";
-
-    {
-        QSqlDatabase db = DataBaseManagement::loadDatabase(databasePath);
-        QSqlQuery selectQuery(db);
 
-        switch (modifier) {
-        case YACReader::NoModifiers:
-            selectQuery.prepare("SELECT ci.number,ci.title,c.fileName,ci.numPages,c.id,c.parentId,c.path,ci.hash,ci.read,ci.isBis,ci.currentPage,ci.rating,ci.hasBeenOpened "
-                                "FROM comic c INNER JOIN comic_info ci ON (c.comicInfoId = ci.id) "
-                                "WHERE UPPER(ci.title) LIKE UPPER(:filter) OR UPPER(c.fileName) LIKE UPPER(:filter) LIMIT :limit");
-            selectQuery.bindValue(":filter", "%%" + filter + "%%");
-            selectQuery.bindValue(":limit", 500); //TODO, load this value from settings
-            break;
-
-        case YACReader::OnlyRead:
-            selectQuery.prepare("SELECT ci.number,ci.title,c.fileName,ci.numPages,c.id,c.parentId,c.path,ci.hash,ci.read,ci.isBis,ci.currentPage,ci.rating,ci.hasBeenOpened "
-                                "FROM comic c INNER JOIN comic_info ci ON (c.comicInfoId = ci.id) "
-                                "WHERE (UPPER(ci.title) LIKE UPPER(:filter) OR UPPER(c.fileName) LIKE UPPER(:filter)) AND ci.read = 1 LIMIT :limit");
-            selectQuery.bindValue(":filter", "%%" + filter + "%%");
-            selectQuery.bindValue(":limit", 500); //TODO, load this value from settings
-            break;
+    beginResetModel();
 
-        case YACReader::OnlyUnread:
-            selectQuery.prepare("SELECT ci.number,ci.title,c.fileName,ci.numPages,c.id,c.parentId,c.path,ci.hash,ci.read,ci.isBis,ci.currentPage,ci.rating,ci.hasBeenOpened "
-                                "FROM comic c INNER JOIN comic_info ci ON (c.comicInfoId = ci.id) "
-                                "WHERE (UPPER(ci.title) LIKE UPPER(:filter) OR UPPER(c.fileName) LIKE UPPER(:filter)) AND ci.read = 0 LIMIT :limit");
-            selectQuery.bindValue(":filter", "%%" + filter + "%%");
-            selectQuery.bindValue(":limit", 500); //TODO, load this value from settings
-            break;
+    qDeleteAll(_data);
 
-        default:
-            QLOG_ERROR() << "not implemented";
-            break;
-        }
+    _data.clear();
 
-        selectQuery.exec();
+    QLOG_ERROR() << "-d2>" << data->size();
```

Comment:
Was this line left here by mistake? It spams error messages every time the search edit's text changes to a non-empty value.

Reply:
Yes, my bad.

Reply:
Should this line be removed or converted to `QLOG_TRACE` or `QLOG_DEBUG`?

Reply:
Removed, I used that log to catch some issues while I was working on this PR but it doesn't have a purpose anymore.

Reply:
Removed in #194.